src/runtime/profiling.cc

```diff
@@ -894,15 +891,21 @@ PackedFunc WrapTimeEvaluator(PackedFunc pf, Device dev, int number, int repeat,
               std::max((min_repeat_ms / (duration_ms / number) + 1), number * golden_ratio));
         }
 
+        int64_t accum_t_nanos = 0;
         // start timing
-        Timer t = Timer::Start(dev);
         for (int j = 0; j < number; ++j) {
+          // call preprocessing function
+          if (f_preproc != nullptr) {
+            f_preproc.CallPacked(args, &temp);
```

Comment:
From experience that this change needs to be reversed.
The **repeat** and **number** parameters are designed for different scenarios. For your scenario, you need to increase the **repeat** and use **number=1**.
Scenario with increasing **number** (or using **min_repeat_ms**) is used when it is problematic to measure single runs - "_The timer is specifically in the outer loop so that we can time functions with a very small runtime or use less precise timers._"

Reply:
Thanks for the clarification, I'll fix it soon.
Unfortunately it looks like there are a lot of misuse of `number` in the codebase, maybe we should create another PR fixing them.

Reply:
> Thanks for the clarification, I'll fix it soon. Unfortunately it looks like there are a lot of misuse of `number` in the codebase, maybe we should create another PR fixing them.
I don't think it matters that much if we are satisfied with the average number and don't need to call f_preproc. So don't be too eager about it.

Reply:
I have reverted `profiling.cc` and updated unittest accordingly.

Reply:
Another concern is whether we should have a `warmup` argument for `time_evaluator` function, I think it's common in all frameworks' profiling tools.

Reply:
> Another concern is whether we should have a `warmup` argument for `time_evaluator` function, I think it's common in all frameworks' profiling tools. 

There is a pre-start call of the function, before starting the logic of measurements.  Do you think that one is not enough? Or warm it up with something else?
```
// skip first time call, to activate lazy compilation components.
pf.CallPacked(args, &temp);
```

Reply:
PyTorch profiler has a `warmup` argument: https://pytorch.org/docs/stable/profiler.html#torch.profiler.schedule, where users can specify a custom number of steps to warm up.

Reply:
I also noticed `ProfileFunction` defined in `profiling.cc` which has a `warmup_iters` argument, which somehow makes me confused: in what case should user select `ProfileFunction` instead of `WrapTimeEvaluator`?

Reply:
Maybe warmup has a place, also we already have skip_first=1 similar to pytorch. 
But this would require changes not only in the C++ part, but also in the C and webasm part. 

For ProfileFunction  plz look original PR with https://github.com/apache/tvm/pull/9553 there are a couple of words about the motivation for adding.

Reply:
Got it, I prefer supporting the `warmup` argument in another PR. Thank you for the discussion!